### PR TITLE
Introduce Domain, EventStore, and EventStreamProcessing modules

### PR DESCRIPTION
**Pull request by twe4ked**
```ruby
class Event
class EventData
class RecordedEvent
class EventHandlers

class Domain::AggregateRepository
class Domain::AggregateRoot

class EventStore::EventBuilder
class EventStore::Postgres::EventStore

class EventStreamProcessing::EventStreamProcessor
class EventStreamProcessing::PersistentSubscription
class EventStreamProcessing::Projector < EventStreamProcessor
class EventStreamProcessing::Reactor < EventStreamProcessor
class EventStreamProcessing::Subscription
class EventStreamProcessing::Postgres::Checkpoint
class EventStreamProcessing::Postgres::Checkpoints
```

```
lib
├── eventory
│   ├── domain
│   │   ├── aggregate_repository.rb
│   │   └── aggregate_root.rb
│   ├── event_store
│   │   ├── postgres
│   │   │   └── event_store.rb
│   │   └── event_builder.rb
│   ├── event_stream_processing
│   │   ├── postgres
│   │   │   ├── checkpoint.rb
│   │   │   └── checkpoints.rb
│   │   ├── event_stream_processor.rb
│   │   ├── persistent_subscription.rb
│   │   ├── projector.rb
│   │   ├── reactor.rb
│   │   └── subscription.rb
│   ├── domain.rb
│   ├── event.rb
│   ├── event_data.rb
│   ├── event_handlers.rb
│   ├── event_store.rb
│   ├── event_stream_processing.rb
│   ├── recorded_event.rb
│   └── version.rb
└── eventory.rb
```

### CODE REVIEW

**Comment by grassdog** [comment]
Should we just merge this in @stevehodgkiss?

File: examples/todo.rb [hunk 1]
```diff
@@ -31,7 +31,7 @@ def complete
   attr_reader :id, :name
 end
 
-class ToDoList < Eventory::AggregateRoot
+class ToDoList < Eventory::Domain::AggregateRoot
```
Comment (stevehodgkiss) [inline examples/todo.rb]: I'd prefer this at the top level, along with the repository.

Comment (grassdog) [reply examples/todo.rb]: Yeah, makes sense. Done in 911f224.

**Comment by twe4ked** [comment]
Thanks for getting this over the line @grassdog. 🙌

**Pull request merged**